### PR TITLE
Firebase dependency in iOS is now optional and can be enabled if remote models are required

### PR DESCRIPTION
**Pull request by acoutts**
This PR makes firebase an optional dependency so that anyone who does not require remote models does not need the additional Firebase dependency.

This is a breaking change because Firebase is now excluded by default for iOS. It must now be enabled with some changes to your Podfile if you wish to use the remote models feature. Because this is new behavior, I bumped the versions for `google_mlkit_image_labeling` and `google_mlkit_object_detection` to their next major version `1.0.0`.

The readme files have been updated accordingly with the required setup steps for iOS to enable remote models.

### CODE REVIEW

File: packages/google_ml_kit/.flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
...uild":true,"dependencies":["google_mlkit_commons"]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"google_mlkit_barcode_scanning","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_commons","dependencies":[]},{"name":"google_mlkit_digital_ink_recognition","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_entity_extraction","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_face_detection","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_image_labeling","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_language_id","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_object_detection","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_pose_detection","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_selfie_segmentation","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_smart_reply","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_text_recognition","dependencies":["google_mlkit_commons"]},{"name":"google_mlkit_translation","dependencies":["google_mlkit_commons"]}],"date_created":"2022-07-11 10:05:51.298252","version":"3.1.0-9.0.pre"}
```
Comment (fbernaly) [inline packages/google_ml_kit/.flutter-plugins-dependencies:1]: This is a generated file; do not edit or check into version control

File: packages/google_mlkit_image_labeling/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: google_mlkit_image_labeling
 description: A Flutter plugin to use Google's ML Kit Image Labeling to detect and extract information about entities in an image across a broad group of categories.
-version: 0.3.0
+version: 1.0.0
```
Comment (fbernaly) [inline packages/google_mlkit_image_labeling/pubspec.yaml:3]: change version to 0.4.0, we are not releasing 1.0.0 yet

**Comment by fbernaly** [comment]
@acoutts: Thanks for the contributions. I left some comments.

File: packages/google_ml_kit/.flutter-plugins-dependencies (hunk 1 above)
Comment (acoutts) [reply packages/google_ml_kit/.flutter-plugins-dependencies]: Done in 17824124197c2b19a0e63ce32244162bf15f371d

File: packages/google_mlkit_image_labeling/pubspec.yaml (hunk 2 above)
Comment (acoutts) [reply packages/google_mlkit_image_labeling/pubspec.yaml]: Done in 72c7a9eeb9f867ae4ecd3733f41e635b8441e3fc and 83d8a5c5fb26e5a9ed2d7f99d5ef0b667c211e9f

**Review by fbernaly — APPROVED** [review]

**Comment by cyberpwnn** [comment]
Still waiting for the flutter dep update, build busted still.

**Pull request merged**